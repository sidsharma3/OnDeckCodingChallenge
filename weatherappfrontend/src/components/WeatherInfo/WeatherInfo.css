* {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
}

.weatherData .textInfo {
    margin: 15px 0 15px 0;
    font-size: 30px;
}

.weatherDataBox {
    height: 60vh;
    width: 40%;
    z-index: 3;
    overflow: hidden;
    background-color: #3395b6;
    margin: auto auto;
    border-radius: 27px;
    box-shadow: 15px 15px 25px -5px rgba(0, 0, 0, 0.5);
}

.cityInputBox {
    padding: 20px;
}

.cityInputBox input {
    outline: none;
    border: none;
    width: 100%;
    font-size: 15px;
    border-radius: 25px;
    padding: 15px;
}


.weatherIcon {
    width: 100px;
    height: 100px;
}