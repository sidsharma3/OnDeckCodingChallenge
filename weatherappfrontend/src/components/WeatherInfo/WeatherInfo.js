import React from 'react';
import "./WeatherInfo.css"

// This componenet is responsible for displaying the weather data
// for the given city and contains the form which is used to select a new city.
const WeatherInfo = ({onEnter, weatherData}) => {
    return (
        <div className="weatherDataBox">

                <div className="cityInputBox">
                    <input 
                    type="text" 
                    placeholder="Type a City Here and Press Enter to get the Weather!"
                    onKeyDown={onEnter}
                    />
                </div>

                {   
                    (weatherData != null && weatherData["status"] === 200)
                    ? 
                    <div className="weatherData">
                        <h2 data-testid="cityAndCountryText" className="location">
                            {`${weatherData["cityName"]}, ${weatherData["country"]}`}
                        </h2>
                        <img 
                            className="weatherIcon"
                            src={`${process.env.REACT_APP_IMG_URL}/${weatherData["icon"]}@2x.png`}
                            alt="Weather Condition Icon"
                        />
                        <h1 className="textInfo">
                            {weatherData["condition"]}
                        </h1>
                        <h1 className="textInfo">
                            {`Current Temp: ${weatherData["temp"]}Â°C`}
                        </h1>
                    </div>
                    : <h1 data-testid="citynotFoundText" > Could Not Find Weather Data on City </h1> 
                }

        </div>
    )
}


export default WeatherInfo;