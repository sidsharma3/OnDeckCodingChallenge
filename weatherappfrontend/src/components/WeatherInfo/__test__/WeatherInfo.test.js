import React from 'react'
import ReactDOM from 'react-dom'
import WeatherInfo from '../WeatherInfo'
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"

// Basic function used as prop for WeatherInfo
const onEnter = (event) => {
    if (event.key === 'Enter') {
        console.log("Enter key Pressed")
    }
}

let weatherDataForRealCity = {
    "status":200,
    "cityName":"Miami",
    "country":"US",
    "icon":"02d",
    "condition":"Clouds",
    "temp":"14.7"
}

let cityNotFoundWeatherData = {
    "status":404,
}

afterEach(cleanup)

it("WeatherInfo component rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
    <WeatherInfo onEnter={onEnter} weatherData={weatherDataForRealCity}/>, div)
})

it("Matches WeatherInfo component Snapshot When City is Found", () => {
    const weatherInfo = renderer.create(
    <WeatherInfo onEnter={onEnter} weatherData={weatherDataForRealCity}/>).toJSON()
    expect(weatherInfo).toMatchSnapshot();
})

it("City Weather Data Found Text Rendered Correctly", () => {
    const { getByTestId } = render(
    <WeatherInfo onEnter={onEnter} weatherData={weatherDataForRealCity}/>)
    expect(getByTestId("cityAndCountryText")).toHaveTextContent("Miami, US")
})

it("Matches WeatherInfo component Snapshot When City is Not Found", () => {
    const weatherInfo = renderer.create(
    <WeatherInfo onEnter={onEnter} weatherData={cityNotFoundWeatherData}/>).toJSON()
    expect(weatherInfo).toMatchSnapshot();
})

it("City Weather Data Not Found Text Rendered Correctly", () => {
    const { getByTestId } = render(
    <WeatherInfo onEnter={onEnter} weatherData={cityNotFoundWeatherData}/>)
    expect(getByTestId("citynotFoundText"))
    .toHaveTextContent("Could Not Find Weather Data on City")
})