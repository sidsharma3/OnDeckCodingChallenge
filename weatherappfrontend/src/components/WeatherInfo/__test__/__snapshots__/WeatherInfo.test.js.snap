// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches WeatherInfo component Snapshot When City is Found 1`] = `
<div
  className="weatherDataBox"
>
  <div
    className="cityInputBox"
  >
    <input
      onKeyDown={[Function]}
      placeholder="Type a City Here and Press Enter to get the Weather!"
      type="text"
    />
  </div>
  <div
    className="weatherData"
  >
    <h2
      className="location"
      data-testid="cityAndCountryText"
    >
      Miami, US
    </h2>
    <img
      alt="Weather Condition Icon"
      className="weatherIcon"
      src="https://openweathermap.org/img/wn/02d@2x.png"
    />
    <h1
      className="textInfo"
    >
      Clouds
    </h1>
    <h1
      className="textInfo"
    >
      Current Temp: 14.7Â°C
    </h1>
  </div>
</div>
`;

exports[`Matches WeatherInfo component Snapshot When City is Not Found 1`] = `
<div
  className="weatherDataBox"
>
  <div
    className="cityInputBox"
  >
    <input
      onKeyDown={[Function]}
      placeholder="Type a City Here and Press Enter to get the Weather!"
      type="text"
    />
  </div>
  <h1
    data-testid="citynotFoundText"
  >
     Could Not Find Weather Data on City 
  </h1>
</div>
`;
